test/test_integration_ssl.rb

```diff
@@ -189,6 +188,33 @@ def test_ssl_run_with_localhost_authority
     end
   end
 
+  def test_ssl_run_with_localhost_authority_in_cluster_mode
+    skip_if :jruby
+
+    cmd = "#{BASE} bin/puma -b \"ssl://0.0.0.0:9292\" -t 5:5 -w 2 test/rackup/self_signed.ru"
```

Comment:
The issue occurs *only* with this setting, when using the env var WEB_CONCURRENCY or when using the puma DSL `workers`  puma starts correctly.
Now since I didn't want to change the `with_server` method I duplicated some code here

Reply:
Looks like you're on the right track 👍

Reply:
I now have three possible ways to handle this scenario:
1. Require the user to use a config file that loads the `localhost` gem so that the methods are available in the `Binder` class.
2. Require the user to use --preload so that the `localhost` gem can be loaded before binding to a socket/port
3. We `require "localhost"` in the `Binder` class. For this we have to check if the gem is available in the current bundle

For options 1 and 2 an additional error message + documentation should be sufficient. These options can also be complementary.

How should I proceed?

Reply:
A fourth option is to find out how the binding works in single process mode and make this work in cluster mode. Don't know how much work this would be...

@nateberkopec Can you please provide me some guidance here?

Reply:
The idea behind this feature was that the user would make the decision to use it by doing `require "localhost"`. As [we have discovered](https://github.com/puma/puma/issues/2818#issuecomment-1172823991), doing that in `config.ru` in cluster mode while not using preload is too late.

So I think that gives that we should go with 1. Most users will probably do something like `require "localhost" if ENV["RACK_ENV"] == "development"`. We can also include in the documentation that 2. works (preload).

Reply:
Re: 3. I don't think Puma should assume bundler is always used and I don't think Puma should try `require "localhost"` unconditionally so I really think 1. is the way to go.

Reply:
Yeah, or rather, begin; require; rescue LoadError; most likely. We can add that to the readme.